### DISCUSSION TITLE
Lifetime in pyclass

### DISCUSSION THREAD

**Discussion by asze17** [PyO3/pyo3#2545] [Questions] [answered]
So I have this code here

```rs
#[pyclass]
pub struct TextSegment<'t> {
    inner: RilTextSegment<'t, Dynamic>
}
```

But I am getting the following error

```
#[pyclass] cannot have generic parameters
```

So my question is why is it not possible to have named lifetime in pyclass, and what can you possibly do to get around it?

**Comment by mejrs** [comment, marked as answer]
You can't have them because Python has no concept of lifetimes. Once you give something to Python, it can keep it around forever, thus it cannot have a lifetime.

**Reply by asze17** [reply]
That sort of what I thought too, thank you.
